### DISCUSSION TITLE
Separate Website and Documentation into their own repos.

### DISCUSSION THREAD

**Discussion by v3ss0n** [fastapi/fastapi#6359] [Questions] [answered]
### Description

* Came back to fastapi after 3 months of not using it.
Sees a lot of Update Release Notes . But actually alll of them are adding new translation.
* Expected  : FastAPI getting some love from the maintainer and flourishing.
* Currently : No new changes , no new features merged , several bugs are still open even though contributors had contributed pull requests for bugs and features . All commits are just translations added and versions are shamelessly bumped

Pushing versions without any new fixes or features is not really honest way of showing the project as active , its deceiving.

### Suggestion 
- Separate Documentation into its own repo , that way documentation can be updated without pushing up versions

**Comment by jonatasoli** [comment]
Is possible you create custom docs in fastapi
https://fastapi.tiangolo.com/id/advanced/extending-openapi/?h=custom#include-the-custom-docs

**Comment by v3ss0n** [comment]
Its not about creating custom docs in fast API 
its about version bumping without any changes but translation in fastAPI repo.

**Comment by rafsaf** [comment]
There are many good reasons to keep docs inside repository with code!

Let's see
- https://github.com/django/django same repo
- https://github.com/pallets/flask  same repo
- https://github.com/sqlalchemy/sqlalchemy/tree/main/doc/build  same repo
- https://github.com/psf/black  same repo
- https://github.com/python-poetry/poetry  same repo
- https://github.com/unbit/uwsgi  another repo https://github.com/unbit/uwsgi-docs

I don't see why would this be better, except for even more distraction for tiangolo. The problem is (very opinionated I would say) practice here that author include those PRs in Release Notes. THIS is uncommon behavior (as far as I know), docs are more often considered as a part of some feature, not a feature itself, and are never included in release notes, maybe with some exceptions like "We added support for language X" that can be considered as useful information.

In other words, release notes are messed up with unnecessary and distracting messages. with no doubt.

But this is for another discussion, and different problem, that there is no real timeline, release process, planning etc.

Like in django:
https://docs.djangoproject.com/en/dev/internals/release-process/

Also note FastAPI is before 1.0.0 and follows semantic versioning, meaning "early stage of the project, breaking changes may happen, go for production at own risk".

**Comment by jacksbox** [comment]
Related to semantic versioning: according to the sem ver specs, doc updates do not increase the version number.
This said, a own repo for the docs is not strictly needed (but would be nice to have: even if the mentioned repos do the same, in the end they are all python repos and the python world imo does a lot of strange things and struggles quite a bit in standardisation), but bumping the version just for doc updates should be omitted

**Comment by v3ss0n** [comment]
> Related to semantic versioning: according to the sem ver specs, doc updates do not increase the version number. This said, a own repo for the docs is not strictly needed (but would be nice to have: even if the mentioned repos do the same, in the end they are all python repos and the python world imo does a lot of strange things and struggles quite a bit in standardisation), but bumping the version just for doc updates should be omitted

Now its even getting worse , all the version bumps are sponsor additions , not  even translations .

**Comment by JarroVGIT** [comment]
> Now its even getting worse , all the version bumps are sponsor additions , not even translations .

This is just not true, and is even easily verifiable when you go to the releases?🤔

**Comment by v3ss0n** [comment]
![image](https://user-images.githubusercontent.com/419606/184712315-e87e496e-35cc-4a19-87e4-ef4ceae958aa.png)
Thats all i can see in 2000 pixel length . What did i miss?

**Comment by JarroVGIT** [comment]
You're screenshot is about commits, not version bumps. See https://github.com/tiangolo/fastapi/releases for the releases, and you can see that not one version bump is just additions of sponsors like you said.

**Comment by Garito** [comment]
And another documentation issues storm
What about those not interested in documentation issues?
My vote to split documentation to another repo

**Comment by v3ss0n** [comment]
Votes doesn't matter in dictatorship.

**Comment by JarroVGIT** [comment]
> Votes doesn't matter in dictatorship.

@v3ss0n: Could you please be a little more respectful and constructive? I’ve noticed this behavior on multiple issues and PRs now from you. It doesn’t add anything, and actually diminishes the value some get out of the community. If you have nothing to contribute, then just say nothing at all.

Thank you.

**Comment by iudeen** [comment]
I see it very clearly bifurcated in release notes into fixes, features, docs and translations. Second, as @JarroVGIT mentioned, sponsor updates or doc updates alone has never gotten a version bump. 

Splitting documentation into a separate is not a great option. Its best to have everything in one place. Also most people would be interested in releases and release notes, and commits should not distract them.

**Comment by tiangolo** [comment, marked as answer]
@v3ss0n your tone and style of communication are critical, demeaning, demanding, and aggressive, and this is not the first time you are showing this behavior here. It is not welcome here, please refrain from continuing with that communication style.

---

The release notes clearly show that each version bump has a reason for the bump, bug fixes, new features, internal changes required in some other way as part of a release (e.g. to unblock Pydantic), etc.

Please verify the things you are saying before posting, as it's misleading and consumes time and effort.

Apart from that, documentation and translations are a very important feature, they belong in the same repo, and contributors that help with that deserve the credit of being "contributors to FastAPI". In fact, those are the people with "most commits" to the repo after me (by a large difference, but still).

If you want to see more progress or progress faster, the way to do it is by helping, as most of the other people in this thread are doing, not by creating new issues demanding more and creating more work for me and for everyone (a new issue to handle). You can help others with their questions in issues, you can review PRs checking that the code is correct, has tests, and actually solves what it is intended to solve. You can help review translations or make translations yourself (that's very important help).

Given all that, I'll close this issue.

---

Thanks to everyone else for the discussions here, in particular, the [FastAPI Experts](https://fastapi.tiangolo.com/fastapi-people/#most-active-users-last-month) like @JarroVGIT and @iudeen, that are always willing to help and foster a kind and welcoming community.
